### PR TITLE
[new release] irmin (19 packages) (3.9.0)

### PR DESCRIPTION
**Pull request by art-w**
Irmin, a distributed database that follows the same design principles as Git

- Project page: <a href="https://github.com/mirage/irmin">https://github.com/mirage/irmin</a>
- Documentation: <a href="https://mirage.github.io/irmin/">https://mirage.github.io/irmin/</a>

### Added

- **irmin-server**
  - Added `irmin-server` package (mirage/irmin#2031, @zshipko)
- **irmin-client**
  - Added `irmin-client` package to connect to `irmin-server` instances (mirage/irmin#2031,
    @zshipko)
- **irmin**
  - Add pretty printers for `Commit`, `Tree`, `Info`, `Status`, `Branch` when
    using `utop` (@metanivek, mirage/irmin#1839)

### Fixed

- **irmin-pack**
  - Fix index integrity check for v3 stores (mirage/irmin#2267, @metanivek)

### Removed

- **irmin-http**
  - Removed `irmin-http` since it is not compatible with generic keys.
    `irmin-grapqhl` or `irmin-server` should be used instead. (mirage/irmin#1902, @zshipko)
- **irmin**
  - Removed stream proofs. We now only have Merkle tree proofs. This simplifies
    the maintenance of that part of the code, as ensuring the correct order of
    cached IO operations was tricky for stream proofs (mirage/irmin#2275, @samoht)

### Changed

- **irmin-git**
  - Moved lower bounds to `git.3.14.0` to use new function (mirage/irmin#2277, @metanivek)

### CODE REVIEW

File: packages/irmin-cli/irmin-cli.3.9.0/opam [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  "irmin-watcher" {>= "0.2.0"}
+  "yaml"          {>= "3.0.0"}
+  "astring"
+  "astring"
+  "cohttp"
+  "cohttp-lwt"
+  "cohttp-lwt-unix"
+  "conduit"
+  "conduit-lwt"
+  "conduit-lwt-unix"
```
Comment (avsm) [inline packages/irmin-cli/irmin-cli.3.9.0/opam:39]:
```suggestion
  "conduit-lwt-unix" {>="6.0.0"}
```

**Comment by avsm** [comment]
Applied a lower bounds fix for irmin-cli

File: packages/irmin-client/irmin-client.3.9.0/opam [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+doc: "https://irmin.org"
+bug-reports: "https://github.com/mirage/irmin/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune" {>= "2.0.0"}
+  "irmin-server" {= version}
+  "irmin-cli" {= version}
+  "ipaddr"
+  "websocket-lwt-unix"
+  "conduit-lwt-unix"
```
Comment (avsm) [inline packages/irmin-client/irmin-client.3.9.0/opam:16]:
```suggestion
  "conduit-lwt-unix" {>="6.0.0"}
```

File: packages/irmin-server/irmin-server.3.9.0/opam [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  "optint" {>= "0.1.0"}
+  "irmin" {= version}
+  "ppx_irmin" {= version}
+  "irmin-pack" {= version}
+  "uri"
+  "fmt"
+  "cmdliner" {>= "1.0.4"}
+  "logs" {>= "0.7.0"}
+  "lwt" {>= "5.4.0"}
+  "conduit-lwt-unix"
```
Comment (avsm) [inline packages/irmin-server/irmin-server.3.9.0/opam:21]:
```suggestion
  "conduit-lwt-unix" {>="6.0.0"}
```

File: packages/irmin-server/irmin-server.3.9.0/opam [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "cmdliner" {>= "1.0.4"}
+  "logs" {>= "0.7.0"}
+  "lwt" {>= "5.4.0"}
+  "conduit-lwt-unix" {>="6.0.0"}
+  "websocket-lwt-unix"
+  "cohttp-lwt-unix"
+  "ppx_blob" {>= "0.7.2"}
+  "digestif" {>= "1.1.4"}
+  "alcotest-lwt" {>= "1.7.0" & with-test}
+  "irmin-watcher" {>= "0.5.0" & with-test}
```
Comment (avsm) [inline packages/irmin-server/irmin-server.3.9.0/opam:27]:
```suggestion
  "irmin-watcher" {>= "0.5.0" & with-test}
  "irmin-test"    {with-test & = version}
```

**Comment by avsm** [comment]
I've applied a number of fixes from the CI failures.

**Comment by tmcgilchrist** [comment]
@art-w is there circular dependency between irmin-server, irmin-server tests and irmin-client? The remaining build errors are from irmin-server test 

`[tests (failed: Library "irmin-client.unix" not found.)](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/dc417d4bf150c6830ea53712af4a20fe494d5916/variant/compilers,4.12,irmin-server.3.9.0,tests)`

**Comment by art-w** [comment]
Thanks a lot @avsm, your changes were very helpful!

@tmcgilchrist > Thanks, indeed, there seems to be an issue with the tests... I've moved them in `irmin-client` to avoid the circular dependency with `irmin-server` :)

**Comment by avsm** [comment]
Thanks! You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

**Pull request merged**